mypy/types.py

```diff
@@ -2925,16 +2926,26 @@ def visit_type_var(self, typ: TypeVarType) -> Type:
         return typ
 
 
+class LocationSetter(TypeTraverserVisitor):
+    def __init__(self, line: int, column: int) -> None:
+        self.line = line
+        self.column = column
+
+    def visit_instance(self, typ: Instance) -> None:
+        typ.line = self.line
+        typ.column = self.column
+        super().visit_instance(typ)
+
+
 def replace_alias_tvars(tp: Type, vars: List[str], subs: List[Type],
                         newline: int, newcolumn: int) -> Type:
     """Replace type variables in a generic type alias tp with substitutions subs
     resetting context. Length of subs should be already checked.
     """
     replacer = InstantiateAliasVisitor(vars, subs)
-    new_tp = tp.accept(replacer)
-    new_tp.line = newline
-    new_tp.column = newcolumn
-    return new_tp
+    newtp = tp.accept(replacer)
+    newtp.accept(LocationSetter(newline, newcolumn))
```

Comment:
To make sure that we don't regress any existing use case, can you also set the `line` and `column` attributes of the return value of the visitor, in case line/column of non-Instance types is important.

Reply:
updated